internal/framework/provider/provider.go

```diff
@@ -153,7 +184,47 @@ func (p *IonosCloudProvider) Configure(ctx context.Context, req provider.Configu
 		return
 	}
 
-	client := s3service.NewClient(accessKey, secretKey, region)
+	cleanedEndpoint := cleanURL(endpoint)
+	version := "DEV"
+
+	newConfig := ionoscloud.NewConfiguration(username, password, token, endpoint)
+	newConfig.UserAgent = fmt.Sprintf(
+		"terraform-provider/%s_ionos-cloud-sdk-go/%s_hashicorp-terraform/%s_terraform-plugin-sdk/%s_os/%s_arch/%s",
+		version, ionoscloud.Version, terraformVersion, meta.SDKVersionString(), runtime.GOOS, runtime.GOARCH, //nolint:staticcheck
+	)
+	if os.Getenv(constant.IonosDebug) != "" {
+		newConfig.Debug = true
+	}
+	newConfig.MaxRetries = constant.MaxRetries
+	newConfig.WaitTime = constant.MaxWaitTime
+	newConfig.HTTPClient = &http.Client{Transport: utils.CreateTransport()}
+	cloudapiClient := ionoscloud.NewAPIClient(newConfig)
```

Comment:
can this be put directly in the bundle struct like the other clients?

Reply:
Can this cloud client configuration specific config be moved to the NewAPIClient?

Reply:
this is because for cloudapi we use the sdk directly, I think changing it would go a bit beyond this PR, maybe you can open a ticket and we can look into it?